### PR TITLE
[skip ci] Changed `0.0.0.0` to `localhost` for universality in CONTIBUTING.md

### PR DESCRIPTION
**Pull request by Priyansi**
* Updated CONTRIBUTING.md
* Changed `0.0.0.0` to `localhost` since `0.0.0.0:<port>` doesn't work for WSL.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -109,13 +109,12 @@ If you modify the code, you will most probably also need to code some tests to e
...
-  - if test function checks distributed configuration, we have to mark the test as `@pytest.mark.distributed` and additional 
-  conditions depending on the intended checks. For example, please see 
-  https://github.com/pytorch/ignite/blob/master/tests/ignite/metrics/test_accuracy.py
-
+  - if test function should run on GPU, please **make sure to add `cuda`** in the test name, e.g. `def test_something_on_cuda()`.
+    Additionally, we may want to decorate it with `@pytest.mark.skipif(not torch.cuda.is_available(), reason="Skip if no GPU")`.
+    For more examples, please see https://github.com/pytorch/ignite/blob/master/tests/ignite/engine/test_create_supervised.py
+  - if test function checks distributed configuration, we have to mark the test as `@pytest.mark.distributed` and additional
+    conditions depending on the intended checks. For example, please see
+    https://github.com/pytorch/ignite/blob/master/tests/ignite/metrics/test_accuracy.py
```
Comment (vfdev-5) [inline CONTRIBUTING.md:117]: Could you please revert the change here.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -313,3 +312,5 @@ python -m http.server <port>
 ```
 
 Then open the browser at `0.0.0.0:<port>` (e.g. `0.0.0.0:1234`) and click to `html` folder.
+
+**NOTE : For WSL, open the browser at localhost:\<port> or 127.0.0.1:\<port> (e.g. localhost:1234 or 127.0.0.1:1234).**
```
Comment (vfdev-5) [inline CONTRIBUTING.md:316]:
```suggestion
**NOTE : For WSL, open the browser at `localhost:<port>` or `127.0.0.1:<port>` (e.g. `localhost:1234` or `127.0.0.1:1234`).**
```

**Review by vfdev-5 — COMMENTED** [review]
Thanks for the @Priyansi ! I left few comments to address. Otherwise, looks good.

File: CONTRIBUTING.md (hunk 1 above)
Comment (Priyansi) [reply CONTRIBUTING.md]: Done!
Actually, I was formatting with pre-commit, and these lines got changed due to that.

File: CONTRIBUTING.md (hunk 2 above)
Comment (Priyansi) [reply CONTRIBUTING.md]: Incorporated the changes. I hope it's alright now!

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -313,3 +313,5 @@ python -m http.server <port>
 ```
 
 Then open the browser at `0.0.0.0:<port>` (e.g. `0.0.0.0:1234`) and click to `html` folder.
+
```
Comment (sdesrozis) [inline CONTRIBUTING.md:316]: Maybe we could just replace `0.0.0.0` by `localhost` and avoid a specific note about WSL ?

Comment (Priyansi) [reply CONTRIBUTING.md:316]: Yeah, `localhost` seems more concise. Okay, I'll change it to that.

Comment (Priyansi) [reply CONTRIBUTING.md:316]: Yeah, `localhost` makes more sense. I'll make the change then.

Comment (Priyansi) [reply CONTRIBUTING.md]: Done!

**Review by sdesrozis — APPROVED** [review]
@Priyansi Thank you LGTM !
